graders/python/python_autograder/pl_execute.py

```diff
@@ -50,7 +65,10 @@ def execute_code(fname_ref, fname_student, include_plt=False,
     with open(fname_ref, 'r', encoding='utf-8') as f:
         str_ref = f.read()
     with open(fname_student, 'r', encoding='utf-8') as f:
-        str_student = f.read()
+        if fname_student[-6:] == '.ipynb':
```

Comment:
Could we use os.path.splitext (https://stackoverflow.com/a/37896418)?  This is probably unlikely, but I could see this breaking if we have a file name that is less than 6 characters.